apps/condo/domains/division/schema/Division.js

```diff
@@ -0,0 +1,143 @@
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.PROTECT' },
+            hooks: {
+                validateInput: async ({ resolvedData, operation, existingItem, addFieldValidationError, context, fieldPath, ...rest }) => {
+                    const responsibleId = resolvedData[fieldPath]
+                    const [responsible] = await OrganizationEmployee.getAll(context, { id: responsibleId })
+                    if (!responsible.role.canBeAssignedAsResponsible) {
+                        addFieldValidationError('Cannot be connected to responsible which does not have `canBeAssignedAsResponsible` ability')
+                    }
+                    let organizationId
+                    if (operation === 'create') {
+                        organizationId = resolvedData.organization
+                    }
+                    if (operation === 'update') {
+                        organizationId = get(existingItem, 'organization')
+                    }
+                    if (organizationId !== responsible.organization.id) {
+                        addFieldValidationError('Cannot be connected to responsible from another organization')
+                    }
+                },
+            },
+        },
+
+        properties: {
+            schemaDoc: 'Properties in service by this division',
+            type: Relationship,
+            ref: 'Property',
+            many: true,
+            hooks: {
+                validateInput: async ({ resolvedData, operation, existingItem, addFieldValidationError, context, fieldPath }) => {
+                    let organizationId
+                    if (operation === 'create') {
+                        organizationId = resolvedData.organization
+                    }
+                    if (operation === 'update') {
+                        organizationId = get(existingItem, 'organization')
+                    }
+                    const [organization] = await Organization.getAll(context, { id: organizationId })
+
+                    const propertyIds = resolvedData[fieldPath]
+                    // Fetch in specified order to be able to test a list of ids in error message
+                    const properties = await Property.getAll(context, { id_in: propertyIds }, { sortBy: ['createdAt_ASC'] })
+                    const propertyIdsFromAnotherOrganization = []
+                    properties.map(property => {
+                        if (property.organization.id !== organization.id) {
+                            propertyIdsFromAnotherOrganization.push(property.id)
+                        }
+                    })
+                    if (propertyIdsFromAnotherOrganization.length > 0) {
+                        addFieldValidationError(`Cannot be connected to following property(s), because they are belonging to another organization(s): ${propertyIdsFromAnotherOrganization.join()}, `)
```

Comment:
Maybe, better to move to constants. Too long error messages. It will be easier to work with constants on client

Reply:
It's to be discussed.
Ideally, yes, we can sync with mobile dev team and ask them to forward error messages in their UI's.
But, this messages are targeted to developers, because they provide technical details.